cdm/misc/src/test/java/ucar/nc2/geotiff/TestGeoTiffPalette.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 1998-2022 University Corporation for Atmospheric Research/Unidata
+ * See LICENSE for license information.
+ */
+package ucar.nc2.geotiff;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ucar.ma2.Array;
+import ucar.ma2.DataType;
+import ucar.nc2.dataset.CoordinateAxis1D;
+import ucar.nc2.dt.GridCoordSystem;
+import ucar.nc2.dt.grid.GridDataset;
+import ucar.nc2.ft2.coverage.*;
+import ucar.nc2.dt.grid.GeoGrid;
+import java.io.File;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.HashMap;
+import java.awt.Color;
+
+/**
+ * Testing the color palette features of GeotiffWriter
+ *
+ * @author WeatherGod
+ * @since 8/12/20224
+ */
+public class TestGeoTiffPalette {
+  private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  @Rule
+  public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+  @Test
+  public void testCreateColorMap() {
+    HashMap<Integer, Color> result = GeotiffWriter.createColorMap(new int[] {10, 3, 145, 4},
+        new String[] {"#00AAff", "#151412", "#DE01aB", "#100ABB"});
+    HashMap<Integer, Color> expected = new HashMap<Integer, Color>();
+    expected.put(10, new Color(0, 170, 255));
+    expected.put(3, new Color(21, 20, 18));
+    expected.put(145, new Color(222, 1, 171));
+    expected.put(4, new Color(16, 10, 187));
+    Assert.assertEquals(expected, result);
+
+    Exception badargs = Assert.assertThrows(IllegalArgumentException.class, () -> {
+      GeotiffWriter.createColorMap(new int[] {10, 3}, new String[] {"#00AAff"});
+    });
+    Assert.assertEquals("flag_values and flag_colors must be of equal length", badargs.getMessage());
+
+    Exception badcolor = Assert.assertThrows(NumberFormatException.class, () -> {
+      GeotiffWriter.createColorMap(new int[] {10, 3}, new String[] {"#00AAff", "Yellow"});
+    });
+    Assert.assertEquals("For input string: \"Yellow\"", badcolor.getMessage());
+  }
+
+  @Test
+  public void testSetColorTable() {
+    HashMap<Integer, Color> colorMap =
+        GeotiffWriter.createColorMap(new int[] {1, 2, 3, 4}, new String[] {"#00AAff", "#151412", "#DE01aB", "#100ABB"});
+    GeotiffWriter writer = new GeotiffWriter("dummy.tif");
+    writer.setColorTable(colorMap, Color.black);
+    int[] resultTable = writer.getColorTable();
+
+    // Explicitly spelling out the RGB channel values [0, 255]
+    // because there were some off-by-one errors in initial development.
+    int[] expectedRGBs = new int[] {0, 170, 255,  // #00AAff
+                                    21, 20, 18,   // #151412
+                                    222, 1, 171,  // #DE01aB
+                                    16, 10, 187}; // #100ABB
```

Comment:
So, spotless wants to re-indent this in a not-ideal manner. Are there any suggested formatting options I can utilize, or am I stuck with these last three lines being out of alignment with the first?